frontend/app/utils/date-utils.ts

```diff
@@ -73,7 +74,20 @@ export function isValidTimeZone(timeZone: string): boolean {
 }
 
 /**
-<<<<<<< HEAD
+ * @param date - Optional date or timestamp to use. If not provided, the current date and time are used.
+ *               Can be a number (milliseconds since epoch), a string (parsable by `new Date()`), or a Date object.
+ * @returns The date formatted in both English ("MMM dd, yyyy") and French ("dd MMM. yyyy").
+ */
+export function dateToLocalizedText(date?: number | string | Date): { dateEn: string; dateFr: string } {
+  const targetDate = new Date(date ?? Date.now());
+  const tzDate = new TZDate(targetDate);
+  return {
+    dateEn: format(tzDate, 'MMM dd, yyyy', { locale: enUS }),
+    dateFr: format(tzDate, 'dd MMM. yyyy', { locale: fr }),
+  };
+}
```

Comment:
Anything that uses `Date.now()` should be normalized to the base timezone, otherwise there is a window of time where the date this returns will not be today's date.

Consider using `getStartOfDayInTimezone(..)` and allow the calling function to pass in the timezone (meaning don't hardcode it in your function).